lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java

```diff
@@ -122,54 +163,83 @@ void testForceCompleteNotClaimed() throws Exception {
 
   @WithAccessId(user = "user-1-1")
   @Test
-  void should_ThrowException_When_TaskIsNotFound() {
-    ThrowingCallable call =
-        () -> taskService.completeTask("TKI:0000000000000000000000000000000000xx");
-    assertThatThrownBy(call).isInstanceOf(TaskNotFoundException.class);
-  }
+  void should_ThrowException_When_CompletingNonExistingTask() {
+    ThrowingCallable call = () -> taskService.completeTask("NOT_EXISTING");
 
-  @WithAccessId(user = "user-1-1")
-  @Test
-  void should_ThrowException_When_UserIsNotAuthorizedOnTask() {
-    ThrowingCallable call =
-        () -> taskService.completeTask("TKI:000000000000000000000000000000000004");
-    assertThatThrownBy(call).isInstanceOf(NotAuthorizedException.class);
+    assertThatThrownBy(call)
+        .isInstanceOf(TaskNotFoundException.class)
+        .extracting(TaskNotFoundException.class::cast)
+        .extracting(TaskNotFoundException::getTaskId)
+        .isEqualTo("NOT_EXISTING");
   }
 
+  // TODO requiredPermissions test missing
   @WithAccessId(user = "user-1-2")
   @Test
-  void should_ThrowException_When_TaskIsInStateReady() {
-    ThrowingCallable call =
-        () -> taskService.completeTask("TKI:000000000000000000000000000000000025");
-    assertThatThrownBy(call).isInstanceOf(InvalidStateException.class);
+  void should_ThrowException_When_UserIsNotAuthorizedOnTask() throws Exception {
+    Task claimedTask = createTaskClaimedByUser().buildAndStore(taskService, "admin");
+
+    ThrowingCallable call = () -> taskService.completeTask(claimedTask.getId());
+
+    AbstractObjectAssert<?, MismatchedWorkbasketPermissionException> exceptionAssert =
+        assertThatThrownBy(call)
+            .isInstanceOf(MismatchedWorkbasketPermissionException.class)
+            .extracting(MismatchedWorkbasketPermissionException.class::cast);
+    exceptionAssert
+        .extracting(MismatchedWorkbasketPermissionException::getCurrentUserId)
+        .isEqualTo("user-1-2");
+    WorkbasketSummary workbasket = claimedTask.getWorkbasketSummary();
+    exceptionAssert
+        .extracting(MismatchedWorkbasketPermissionException::getWorkbasketId)
+        .isEqualTo(workbasket.getId());
+    exceptionAssert.extracting(
+        MismatchedWorkbasketPermissionException
+            ::getRequiredPermissions); // check for invalid permission
```

Comment:
This is long and hard to read :/ 
Let's find a simpler way to do the same thing :)

This might help: https://joel-costigliola.github.io/assertj/core-8/api/org/assertj/core/api/Assertions.html#catchThrowableOfType-org.assertj.core.api.ThrowableAssert.ThrowingCallable-java.lang.Class-

Alternatives could be some suggestions from this question: https://stackoverflow.com/questions/64541192/assertthatthrownby-check-field-on-custom-exception